### PR TITLE
[MPS] Work around MPSGraph constant pad dropping input above 2^16 inner extent

### LINKED ISSUE #194922: F.pad(mode="constant") with temporal-axis padding silently corrupts ~80% of elements for large 5D tensors on MPS
### 🐛 Describe the bug

`F.pad` with `mode="constant"` along the **temporal (dim 2) axis of a 5D tensor** silently writes wrong values into most of the output tensor on the MPS backend when the padded spatial extent is large. No error is raised; corrupted values look statistically plausible (same distribution), so downstream models silently produce garbage.

This silently corrupted a production video-VAE encoder (ComfyUI MiniMax-H3 VAE) above ~224x480 resolution: latents looked plausible but decoded to flat "tan" videos.

### Environment

- PyTorch 2.13.0, macOS arm64, Apple M4 Max
- Reproduces in fp32 and fp16 (fp16 additionally produces NaNs in the padded output)

### Minimal repro

```python
import torch
import torch.nn.functional as F

torch.manual_seed(3)
x = torch.randn(1, 3, 9, 256, 544)          # 5D, fp32
pad = (0, 0, 0, 0, 2, 0)                    # temporal front-only

a = F.pad(x.to("mps"), pad).cpu()           # MPS
b = F.pad(x, pad)                           # CPU
print(torch.equal(a, b))                    # False (~80% of elements differ)
print(float((a - b).abs().max()))           # ~5-6 (input range is +/-4)
```

### Characterization

Elementwise compare vs CPU (`torch.equal`), `pad=(0,0,0,0,2,0)`, fp32, T=9, C=3:

| spatial size | elems/frame (padded) | bad fraction |
|---|---|---|
| 96x224 | 22K | 0.0% |
| 128x288 | 37K | 0.0% |
| 160x352 | 57K | 0.0% |
| 192x416 | 81K | **74.8%** |
| 224x480 | 109K | **58.4%** |
| 252x540 | 137K | **80.3%** |
| 256x544 | 141K | **79.7%** |
| 352x768 | 272K | **80.5%** |

- Threshold tracks elements per spatial frame, near 2^16 = 65,536
- Damage pattern (256x544): output frame 0 (the newly padded frame) is 100% correct; frame 1 is 4.6% bad; frames 2+ are 94.06% bad. Exactly 131,072 (2^17) elements wrong per frame, 8,192 (2^13) correct — the power-of-2 geometry suggests a tiling/grid bug in the pad kernel
- Reproduces with: back-only `(0,0,0,0,0,2)`, symmetric `(0,0,0,0,1,1)`, any `value`, C=128, T=2..17, fp16 (NaNs)
- Does **not** reproduce with spatial-only constant padding `(2,0,0,0,0,0)` even at 352x768, nor with `mode="reflect"`/`"replicate"` on any axis
- A `conv3d` consuming the corrupted pad output is itself correct; padding on CPU then moving to MPS gives bit-exact conv results
- `.clone()`, `.contiguous()`, or inserting ops between pad and consumer do not change the corrupted values (values are wrong at pad time, not a layout issue)

### Impact

Any MPS user of 3D-conv models with causal/temporal zero-padding at >= ~256px spatial sizes (video VAE encoders, video diffusion) gets silent data corruption. Because the threshold is size-dependent and the pattern is model-code-dependent, it presents to users as "model X breaks at 720p on Mac" rather than as a framework bug.

### Workaround

Replace temporal constant-pad with `torch.cat` of a zero slab (bit-exact vs CPU, verified), or perform the pad on CPU.

Possibly related class of bug: #188359 (silently incorrect conv2d output for kernel size >= 256 on MPS).

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

### PR DESCRIPTION
**Pull request by MohammadSadeghSalehi**
Fixes #194922.

`MPSGraph`'s `padTensor` with `MPSGraphPaddingModeConstant` writes the constant fill value **in place of the input data** once the product of the two innermost dimensions reaches 65536. The output has the correct shape and a correct padded region, so nothing downstream notices; users get plausible-looking tensors that are silently wrong. This reached the reporter of #194922 as a video VAE that decoded to flat colour above 224x480.

### This is not a PyTorch kernel bug

Reproduced against `MPSGraph` directly, with no PyTorch in the process (standalone Objective-C++, attached to the Apple report):

```
  N=1 C=1 T=4 H=64  W=1023  H*W=65472   bad=0/392832       (0.0000)
  N=1 C=1 T=4 H=64  W=1024  H*W=65536   bad=262144/393216  (0.6667)
  N=1 C=1 T=4 H=256 W=256   H*W=65536   bad=262144/393216  (0.6667)
  N=1 C=1 T=4 H=64  W=2048  H*W=131072  bad=524288/786432  (0.6667)
```

`bad` equals the input element count exactly, and inspecting the values shows the fill constant written where data should be (196992/262144 zeros at the threshold, 524288/524288 at twice it). Filed with Apple as **FB24527244**.

PyTorch cannot fix the kernel, but it can avoid handing `MPSGraph` the failing configuration, which is what this PR does.

### Guard condition

The corruption needs the inflated-padding path, `ndims > padding_dim + 1`. Rank 4 with an identical 65536 inner extent is clean because `4 > 4` is false; ranks 5 and 6 both corrupt. The threshold is exactly 65536 and is independent of how the extent splits, so 64x1024 and 256x256 both fail.

This routes that configuration to `at::native::constant_pad_nd`, already the fallback for `pad.size() > 6`, and bit-exact against CPU on every affected shape. Everything below the threshold keeps the fast `MPSGraph` path.

### Before and after

Built from this branch point, toggling only `Pad.mm`:

| shape | inner extent | before | after |
|---|---|---|---|
| (1,1,4,64,1023) | 65472 | 0.0000 | 0.0000 |
| (1,1,4,64,1024) | 65536 | **0.6667** | 0.0000 |
| (1,1,4,64,2048) | 131072 | **0.6667** | 0.0000 |
| (1,3,9,256,544) | 139264 | **0.8021** | 0.0000 |

float32 and float16 both. Note this reproduces on **current main**, not only on the 2.13.0 the issue was filed against.

### Test

`test_constant_pad_large_inner_extent` brackets the threshold at 64x1023 against 64x1024 in float32 and float16. The tight pair is deliberate: it fails if the guard is removed, and also if the guard is widened enough to lose the fast path unnecessarily.

Local run: the new test passes, and `-k pad` gives 127 passed. Four `TestSDPA::test_prefill_attention_*` tests fail in my environment with `Failed to create function state object for: prefill_attention_mpp_...`, because I had to build with `CAN_COMPILE_METAL_40=NO` — my Xcode's Metal toolchain rejects `mpp::tensor_ops` members used in `MppAttention.h`. Unrelated to this change and expected to pass in CI.

### Open question for reviewers

Should this be gated on an OS version so it lifts automatically once Apple ships a fix? I have kept it unconditional because the affected range is not yet known, and silently wrong numerics seem worse than losing the fast path on some shapes. Happy to change it.

### CODE REVIEW

**Comment by MohammadSadeghSalehi** [comment]
Superseded by #195147, which is the better fix. Closing.

Mine guarded `constant_pad_nd_mps` and routed the failing shapes to `at::native::constant_pad_nd`, which gives up the MPSGraph path and only covers constant padding. @robertomeroni instead folds the leading dims into one batch dim inside `pad_out_template`, pads at rank <= 4 where MPSGraph is correct, and reshapes back in-graph. That keeps the fast path and also fixes `reflection_pad3d` and `replication_pad3d`, forward and backward, which are equally affected and which I had not checked.

Their boundary condition is also sharper than the one I posted: `max(output last dim, product of output dims after the padded axis) >= 65536`, where I had only tested the product of the two innermost dims.

For the record, the upstream fault is still Apple's, filed as FB24527244 with a standalone MPSGraph repro attached. #195147 works around it correctly, but `MPSGraph` still writes the constant fill value in place of the input above the threshold, so it is worth keeping in mind if the rank <= 4 path ever changes.
